ProjFS.Mac/PrjFSKextTests/KextAssertIntegration.m

```diff
@@ -0,0 +1,90 @@
+#import "KextAssertIntegration.h"
+#include <pthread.h>
+
+extern void Assert(const char* file, unsigned line, const char* expression);
+
+static pthread_key_t s_currentTestCaseKey;
+
+static PFSKextTestCase* CurrentTestCase()
+{
+    return (__bridge PFSKextTestCase*)pthread_getspecific(s_currentTestCaseKey);
```

Comment:
What does `__bridge` do?

Reply:
It's used for casting between pointers to Objective-C objects and regular pointers. (typically `void*` as here, or Core Foundation types for "toll-free bridged" classes, such as `NSString` <-> `CFStringRef`.)

The reason is that Objective-C pointers use ARC, automatic reference counting (unless this is disabled via a compiler flag). This means that `PFSKextTestCase*` is effectively a "smart" pointer comparable to C++'s `std::shared_ptr<>`. When casting to an opaque type such as `void*`, it needs to be made clear to the compiler whether ownership of the object is transferred, whether an additional reference should be taken, or whether the opaque pointer is effectively a weak pointer (as it is here).

A more detailed explanation of ARC and bridged casts [can be found in the llvm/clang documentation](https://clang.llvm.org/docs/AutomaticReferenceCounting.html#bridged-casts).

Reply:
It's used for casting between pointers to Objective-C objects and regular pointers. (typically `void*` as here, or Core Foundation types for "toll-free bridged" classes, such as `NSString` <-> `CFStringRef`.)

The reason is that Objective-C pointers use ARC, automatic reference counting (unless this is disabled via a compiler flag). This means that `PFSKextTestCase*` is effectively a "smart" pointer comparable to C++'s `std::shared_ptr<>`. When casting to an opaque type such as `void*`, it needs to be made clear to the compiler whether ownership of the object is transferred, whether an additional reference should be taken, or whether the opaque pointer is effectively a weak pointer (as it is here).

A more detailed explanation of ARC and bridged casts [can be found in the llvm/clang documentation](https://clang.llvm.org/docs/AutomaticReferenceCounting.html#bridged-casts).